base/reduce.jl

```diff
@@ -360,15 +360,19 @@ sum(a::AbstractArray{Bool}) = countnz(a)
 Returns the sum of all array elements, using the Kahan-Babuska-Neumaier compensated
 summation algorithm for additional accuracy.
 """
-function sum_kbn{T<:AbstractFloat}(A::AbstractArray{T})
+function sum_kbn(A::Any)
+    T = _default_eltype(typeof(A))
     c = r_promote(+, zero(T)::T)
-    if isempty(A)
+    i = start(A)
+
```

Comment:
I don't mean to encourage lack of readability, so sorry for a better wording above. Still, this seems pretty readable to me without the empty lines, and also matches the style used on the rest of the file and a bunch of functions like this one.